xdsl/dialects/x86/assembly.py

```diff
@@ -78,7 +78,7 @@ def memory_access_str(
     register: AssemblyInstructionArg, offset: IntegerAttr | None
 ) -> str:
     register_str = assembly_arg_str(register)
-    if offset is not None:
+    if offset is not None and offset.value.data != 0:
```

Comment:
can you please open a separate PR for this, along with a small change to the x86 ops assembly filecheck test?